lib/database_validations/lib/validations.rb

```diff
@@ -6,6 +6,28 @@ module Validations
       alias_method :validate, :valid?
     end
 
+    class_methods do
+      def create_or_find_by!(attributes, &block)
+        super
+      rescue ActiveRecord::RecordInvalid => e
+        rescue_create_or_find_by_uniqueness_exception(e, attributes)
+      end
+
+      def create_or_find_by(attributes, &block)
+        super
+      rescue ActiveRecord::RecordInvalid => e
+        rescue_create_or_find_by_uniqueness_exception(e, attributes)
+      end
+
+      private
+
+      def rescue_create_or_find_by_uniqueness_exception(err, attributes)
+        raise err unless err.is_a?(ActiveRecord::NotUnique) || err&.cause&.is_a?(ActiveRecord::RecordNotUnique)
```

Comment:
**Question:** Do we need `err.is_a?(ActiveRecord::NotUnique)`? I thought it's processed in `super` already.

Reply:
**Note:**

Never knew about `cause` available on the ActiveRecord errors :+1: But do we need `err&`, won't it be not nil always in your case?